src/VirtoCommerce.CatalogModule.Web/Controllers/Api/CatalogModuleProductsController.cs

```diff
@@ -86,6 +86,8 @@ public async Task<ActionResult<CatalogProduct[]>> GetByCodes([FromRoute] string
         ///<param name="respGroup">Response group.</param>
         [HttpGet]
         [Route("")]
+        [ProducesResponseType(typeof(IEnumerable<CatalogProduct>), StatusCodes.Status200OK)]
+        [ProducesResponseType(typeof(void), StatusCodes.Status404NotFound)]
         public async Task<ActionResult> GetProductByIds([FromQuery] List<string> ids, [FromQuery] string respGroup = null)

```

Comment:
```suggestion
        public async Task<ActionResult<CatalogProduct[]>> GetProductByIds([FromQuery] List<string> ids, [FromQuery] string respGroup = null)
```

Reply:
I think the whole point was **not** to return a strongly typed response, but serialize it manually in action's body to avoid performance issue with Newtonsoft.Json.  See the comment I linked in the description and the issue referenced in that comment.

Reply:
The point was to serialize manually, the return type in method signature doesn't matter.